File: als-server/jvm/src/main/scala/org/mulesoft/als/server/lsp4j/extension/UpdateConfigurationParams.java
package org.mulesoft.als.server.lsp4j.extension;

import org.mulesoft.als.configuration.TemplateTypes;

import java.util.Map;

public class UpdateConfigurationParams {

    private Map<String, AlsFormattingOptions> formattingOptions;

    private String templateType;

    private Map<String, Object> genericOptions;

    public UpdateConfigurationParams(Map<String, AlsFormattingOptions> updateFormatOptionsParams) {
        this.formattingOptions = updateFormatOptionsParams;
    }

    public UpdateConfigurationParams(Map<String, AlsFormattingOptions> updateFormatOptionsParams, String templateType) {
        this.formattingOptions = updateFormatOptionsParams;
        this.templateType = templateType;
    }

    public Map<String, AlsFormattingOptions> getUpdateFormatOptionsParams() {
        return formattingOptions;
    }

    public Map<String, Object> getGenericOptions() {
        return genericOptions;
    }

    public String getTemplateType() {
        return templateType;
    }

    public void setTemplateType(String templateType) {
        this.templateType = templateType;
    }
}

File: als-server/shared/src/main/scala/org/mulesoft/als/server/feature/configuration/UpdateConfigurationParams.scala
package org.mulesoft.als.server.feature.configuration

import org.mulesoft.als.configuration.TemplateTypes
import org.mulesoft.lsp.configuration.FormattingOptions

import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}

case class UpdateConfigurationParams(
    updateFormatOptionsParams: Option[Map[String, FormattingOptions]],
    genericOptions: Map[String, Any] = Map.empty,
    templateType: String = TemplateTypes.FULL,
    prettyPrintSerialization: Boolean = false
)

@JSExportAll
@JSExportTopLevel("GenericOptionKeys")
object GenericOptionKeys {
  val KeepTokens = "keepTokens"
}
